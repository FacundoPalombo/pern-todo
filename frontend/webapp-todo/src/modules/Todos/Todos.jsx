import React from "react";
import { Todo } from "./components";
export const Todos = ({ todos = [] }) => {
  return (
    <article>
      {todos.map((t) => (
        <Todo description={t.description} key={t.id} />
      ))}
    </article>
  );
};
