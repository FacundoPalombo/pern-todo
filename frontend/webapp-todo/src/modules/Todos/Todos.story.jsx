import React from "react";
import { Todos } from "./Todos";

const test_todos = [
  { description: "Do homework", id: "1" },
  { description: "lorem ipsum", id: "2" },
  { description: "dolor sit amet", id: "3" },
];

export const withTodos = () => <Todos todos={test_todos}></Todos>;

export const withoutTodos = () => <Todos />;

export default {
  title: "Todos",
  component: Todos,
};
