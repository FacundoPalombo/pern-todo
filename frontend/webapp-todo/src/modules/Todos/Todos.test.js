import React from "react";
import { render } from "@testing-library/react";
import { Todos } from "./Todos";

describe("GIVEN Todos component", () => {
  it("THEN should render", () => {
    const { asFragment } = render(<Todos />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("THEN should render with todos when todos is provided", () => {
    const mockedTodos = [
      { id: "0", description: "test1" },
      { id: "2", description: "test2" },
      { id: "3", description: "test4" },
    ];
    const { getByText } = render(<Todos todos={mockedTodos} />);
    expect(getByText("test1")).not.toBeNull();
    expect(getByText("test2")).not.toBeNull();
    expect(getByText("test4")).not.toBeNull();
  });
});
