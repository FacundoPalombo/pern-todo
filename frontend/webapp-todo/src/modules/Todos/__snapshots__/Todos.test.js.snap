// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIVEN Todos component THEN should render 1`] = `
<DocumentFragment>
  <article />
</DocumentFragment>
`;
